### PR TITLE
Documentation freeze

### PR DESCRIPTION
**Pull request by EmyrClement**
- Lepton id/iso/trigger systematics
- JER systematic
- QCD scale
- Start pdf uncertainties

- Pick bins
I removed the n events criteria from the final bin, so it's not merged with previous bin as long as purity and stability are ok.  I did this while debugging another issue (see changes to prepare_dps.sh).  We should add this back in, or discuss how to go about this.

### CODE REVIEW

File: src/cross_section_measurement/make_control_plots_fromTrees.py [hunk 1]
```diff
@@ -142,20 +168,34 @@ def make_plot( channel, x_axis_title, y_axis_title,
...
             print 'Overall scale : ',dataDrivenQCDScale
-            qcd_from_data.Scale( dataDrivenQCDScale )
+            qcd_from_data.Scale( dataDrivenQCDScale.nominal_value )
             signalToControlScale = n_qcd_predicted_mc_signal / n_qcd_control_region
             dataToMCscale = n_qcd_control_region / n_qcd_predicted_mc_control
             print "Signal to control :",signalToControlScale
             print "QCD scale : ",dataToMCscale
     else:
         qcd_from_data = signal_region_hists['QCD']
+        print qcd_from_data.Integral()
```
Comment (kreczko) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: print?

File: src/cross_section_measurement/make_control_plots_fromTrees.py [hunk 2]
```diff
@@ -142,20 +168,34 @@ def make_plot( channel, x_axis_title, y_axis_title,
...
+
+        n, error = control_region_hists['QCD'].integral(0,nBins+1,error=True)
+        n_qcd_predicted_mc_control = ufloat( n, error)
+
+        n, error = qcd_from_data.integral(0,nBins+1,error=True)
+        n_qcd_control_region = ufloat( n, error)
+
+        print n_qcd_predicted_mc_signal
+        print n_qcd_predicted_mc_control
+        print n_qcd_control_region
```
Comment (kreczko) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: are these really needed?

File: src/cross_section_measurement/make_control_plots_fromTrees.py [hunk 3]
```diff
@@ -11,12 +11,37 @@
...
+                mcHist = histogram.Clone()
+            else:
+                mcHist += histogram
+    dataValues = list(dataHist.y())
+    mcValues = list(mcHist.y())
+
+    weights = [ data / mc for data, mc in zip(dataValues, mcValues)]
+    print list(dataHist.x())
+    print list(dataHist.xedges())
+    print weights
```
Comment (kreczko) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: more prints

File: experimental/BLTUnfold/produceUnfoldingHistograms.py [hunk 4]
```diff
@@ -315,27 +317,30 @@ def main():
                 # Fill histograms
                 #
                 if not options.donothing:
-                    tree.Draw('(EventWeight * %.4f)' % measurement_config.luminosity_scale,'1',hist=eventWeight)
+                    # tree.Draw('(EventWeight * %.4f)' % measurement_config.luminosity_scale,'1',hist=eventWeight)
                     # 1D
-                    tree.Draw(genVariable_particle,genWeight+'*'+genSelection,hist=truth)
-                    tree.Draw(genVariable_particle,genWeight+'*'+genSelectionVis,hist=truthVis)
+                    print 'True'
```
Comment (kreczko) [inline experimental/BLTUnfold/produceUnfoldingHistograms.py]: ```True```

File: experimental/BLTUnfold/produceUnfoldingHistograms.py [hunk 5]
```diff
@@ -315,27 +317,30 @@ def main():
...
+                        tree.Draw(genVariable_parton,genWeight+'*'+genSelection,hist=truth_parton, nentries=nEntries)
+                    print "Measured"
+                    tree.Draw(recoVariable,offlineWeight+'*'+offlineSelection,hist=measured, nentries=nEntries)
+                    tree.Draw(recoVariable,offlineWeight+'*'+offlineSelection,hist=measuredVis, nentries=nEntries)
+                    tree.Draw(recoVariable,offlineWeight+'*'+fakeSelection,hist=fake, nentries=nEntries)
                     # 2D
-                    tree.Draw(recoVariable+':'+genVariable_particle,offlineWeight+'*'+offlineSelection,hist=response)
-                    tree.Draw(recoVariable+':'+genVariable_particle,offlineWeight+'* ('+offlineSelection+'&&'+genSelection +')',hist=response_without_fakes)
-                    tree.Draw(recoVariable+':'+genVariable_particle,offlineWeight+'*'+fakeSelection,hist=response_only_fakes)
+                    print 'Response'
```
Comment (kreczko) [inline experimental/BLTUnfold/produceUnfoldingHistograms.py]: 'Question'

File: experimental/BLTUnfold/produceUnfoldingHistograms.py [hunk 6]
```diff
@@ -315,27 +317,30 @@ def main():
...
+                    print 'True'
+                    tree.Draw(genVariable_particle,genWeight+'*'+genSelection,hist=truth, nentries=nEntries)
+                    tree.Draw(genVariable_particle,genWeight+'*'+genSelectionVis,hist=truthVis, nentries=nEntries)
                     if genVariable_parton != None:
-                        tree.Draw(genVariable_parton,genWeight+'*'+genSelection,hist=truth_parton)
-                    tree.Draw(recoVariable,offlineWeight+'*'+offlineSelection,hist=measured)
-                    tree.Draw(recoVariable,offlineWeight+'*'+offlineSelection,hist=measuredVis)
-                    tree.Draw(recoVariable,offlineWeight+'*'+fakeSelection,hist=fake)
+                        tree.Draw(genVariable_parton,genWeight+'*'+genSelection,hist=truth_parton, nentries=nEntries)
+                    print "Measured"
```
Comment (kreczko) [inline experimental/BLTUnfold/produceUnfoldingHistograms.py]: indeed

File: src/cross_section_measurement/make_control_plots_fromTrees.py [hunk 7]
```diff
@@ -11,12 +11,37 @@
...
 # latex, font, etc
 setup_matplotlib()
 
 title_template = '$%.0f$ pb$^{-1}$ (%d TeV)'
 
+def getPUWeights(histograms_to_draw, histogram_lables) :
+    print 'In get pu weights'
+    # hists = dict(zip(histogram_lables, histograms_to_draw))
+    hists = {}
+    print histogram_lables
```
Comment (kreczko) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: Please remove

File: src/cross_section_measurement/make_control_plots_fromTrees.py [hunk 8]
```diff
@@ -11,12 +11,37 @@
...
+    dataHist = None
+    mcHist = None
+    for label, histogram in zip(histogram_lables, histograms_to_draw):
+        print label
```
Comment (kreczko) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: same here

File: src/cross_section_measurement/make_control_plots_fromTrees.py (hunk 2 above)
Comment (EmyrClement) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: They are used in calculating the QCD scale factors?  I removed the print statements.

File: src/cross_section_measurement/make_control_plots_fromTrees.py (hunk 3 above)
Comment (EmyrClement) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: These print out the pu weights for use elsewhere (not used anywhere yet though).

File: experimental/BLTUnfold/produceUnfoldingHistograms.py (hunk 4 above)
Comment (EmyrClement) [inline experimental/BLTUnfold/produceUnfoldingHistograms.py]: Removed

File: experimental/BLTUnfold/produceUnfoldingHistograms.py (hunk 5 above)
Comment (EmyrClement) [inline experimental/BLTUnfold/produceUnfoldingHistograms.py]: Removed

File: src/cross_section_measurement/make_control_plots_fromTrees.py (hunk 7 above)
Comment (EmyrClement) [inline src/cross_section_measurement/make_control_plots_fromTrees.py]: Done

**Pull request merged**